### PR TITLE
Routing readme updates

### PR DESCRIPTION
**Pull request by tTakeda09**
**Sample**

**Summary**

This PR is to improve the Readme
- Change the sample name to start with a verb (was requested by DevEx)
- Add "How to use sample" for the information about how users can see the functionality
- Make "How it works" focused on how the sample is done to show the functionality

**Tests**

**ArcGIS Maps SDK Version**

### CODE REVIEW

File: samples_project/Assets/SampleViewer/Samples/Routing/readme.md [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
 
 ![Routing](routing.png)
 
+## How to use the sample
+
+This sample uses Esri's [routing service's REST API](https://developers.arcgis.com/rest/network/api-reference/overview-of-network-analysis-services.htm) to query the closest route along the road network between two points. This service uses routing operations associated with your API Key. You can learn more about [API keys](https://developers.arcgis.com/documentation/mapping-apis-and-services/security/api-keys/) and [Accounts](https://developers.arcgis.com/documentation/mapping-apis-and-services/deployment/accounts/) in the _Mapping APIs and location services_ guide.
+
+1. Click on the **RouteManager** game object under **ArcGISMap** in the **Hierarchy** window.
+2. Set your API key in the **Inspector** window.
+3. While holding shift, left click on the map twice. (This sample is only setup to work with mouse and keyboard.)
```
Comment (avr757) [inline samples_project/Assets/SampleViewer/Samples/Routing/readme.md:13]:
```suggestion
3. While holding shift, left click on the map twice. (This sample is only set up to work with mouse and keyboard.)
```

**Review by avr757 — CHANGES_REQUESTED** [review]

**Review by Jade-JadeH — APPROVED** [review]

File: samples_project/Assets/SampleViewer/Samples/Routing/readme.md [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
 
 ![Routing](routing.png)
 
+## How to use the sample
+
+This sample uses Esri's [routing service's REST API](https://developers.arcgis.com/rest/network/api-reference/overview-of-network-analysis-services.htm) to query the closest route along the road network between two points. This service uses routing operations associated with your API Key. You can learn more about [API keys](https://developers.arcgis.com/documentation/mapping-apis-and-services/security/api-keys/) and [Accounts](https://developers.arcgis.com/documentation/mapping-apis-and-services/deployment/accounts/) in the _Mapping APIs and location services_ guide.
+
+1. Click on the **RouteManager** game object under **ArcGISMap** in the **Hierarchy** window.
+2. Set your API key in the **Inspector** window.
+3. While holding shift, left click on the map twice. (This sample is only set up to work with mouse and keyboard.)
```
Comment (tTakeda09) [inline samples_project/Assets/SampleViewer/Samples/Routing/readme.md:13]:
```suggestion
1. Open the **Routing** level.
2. Click on the **RouteManager** game object under **ArcGISMap** in the **Hierarchy** window.
3. Set your API key in the **Inspector** window.
4. While holding shift, left click on the map twice. (This sample is only set up to work with mouse and keyboard.)
```

Comment (tTakeda09) [reply samples_project/Assets/SampleViewer/Samples/Routing/readme.md:13]: @Jade-JadeH @avr757 I'm not thinking of changing the sample level name but the sample name on doc.
So I think it's better to include the name of the sample level. What do you think?

**Review by avr757 — APPROVED** [review]

**Review by Jade-JadeH — APPROVED** [review]
looks good to me.

File: samples_project/Assets/SampleViewer/Samples/Routing/readme.md [hunk 3]
```diff
@@ -1,15 +1,24 @@
...
-4. Raycasts are used to determine the elevation at each breadcrumb's position to account for elevation.
+1. The `HTTPClient` is required for using the [routing service's REST API](https://developers.arcgis.com/rest/network/api-reference/overview-of-network-analysis-services.htm). 
+2. Use the [direct request](https://developers.arcgis.com/rest/network/api-reference/route-synchronous-service.htm) to request and to obtain the routing result.
+3. Use [Raycast](https://docs.unity3d.com/ScriptReference/Physics.Raycast.html) to determine the elevation at each breadcrumb's position to account for elevation. (Enable the mesh collider from the **ArcGIS Map** component to use the Raycast.)
+4. Use Unity's [Line Renderer component](https://docs.unity3d.com/Manual/class-LineRenderer.html) to draw the route.
```
Comment (tTakeda09) [inline samples_project/Assets/SampleViewer/Samples/Routing/readme.md:21]:
```suggestion
1. The `HTTPClient` is used to query the [routing service's REST API](https://developers.arcgis.com/rest/network/api-reference/overview-of-network-analysis-services.htm). 
2. The sample makes a [direct request](https://developers.arcgis.com/rest/network/api-reference/route-synchronous-service.htm) to obtain the routing result.
3. The [Raycast](https://docs.unity3d.com/ScriptReference/Physics.Raycast.html) is used to determine the elevation at each breadcrumb's position to account for elevation. (Mesh colliders need to be enabled in the **ArcGIS Map** component to use the Raycast.)
4. Unity's [Line Renderer component](https://docs.unity3d.com/Manual/class-LineRenderer.html) are used to visualize the route segments between pairs of breadcrumbs.
```

**Pull request merged**